src/huggingface_hub/hf_api.py

```diff
@@ -10346,8 +10367,8 @@ def run_uv_job(
         if namespace is None:
             namespace = self.whoami(token=token)["name"]
 
-        if script.startswith("http://") or script.startswith("https://"):
-            # Direct URL execution - no upload needed
+        if script.startswith("http://") or script.startswith("https://") or not script.endswith(".py"):
+            # Direct URL execution or command - no upload needed
```

Comment:
maybe more robust to check for the presence of a file here because the script could be any file type, like a `.sh` file, not just one ending in `.py`.
```suggestion
        is_url = script.startswith("http://") or script.startswith("https://")
        if is_url or not os.path.isfile(script):
            # Direct URL execution or command - no upload needed
```